.github/workflows/Deploy_web_site_with_ssh.yml

```
@@ -0,0 +1,32 @@
+name: Deploy_web_site_with_ssh
+
+on:
+  push:
+    branches:
+      - master
+  workflow_dispatch:
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:   
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Change BASE_PATH
+        run: yq -i '.JB.BASE_PATH = "${{ vars.CGAL_WEB_SITE_URL }}"' _config.yml
+      - name: Remove demo/ (temporarily)
+        run: rm -rf demo/
+      - name: Build with Jekyll
+        uses: actions/jekyll-build-pages@v1
+        with:
+          source: ./
+          destination: ./_site
+      - name: install ssh keys
+        run: |
+          install -m 600 -D /dev/null ~/.ssh/id_rsa
+          echo "${{ secrets.SSH_PRIVATE_KEY }}" > ~/.ssh/id_rsa
+          ssh-keyscan -H ${{ vars.SSH_HOST }} > ~/.ssh/known_hosts
+      - name: rsync
+        run: rsync -avz ./_site/ ${{ vars.SSH_USER }}@${{ vars.SSH_HOST }}:/home/${{ vars.SSH_USER }}/public_html/cgal-web
```

- The destination directory should be configurable.
- Rename the variables and secrets to add `CGAL_WEB_SFTP_` as a prefix.
- And actually the user name should be a secret, and not a variable.

```suggestion
          echo "${{ secrets.CGAL_WEB_SFTP_SSH_PRIVATE_KEY }}" > ~/.ssh/id_rsa
          ssh-keyscan -H ${{ vars.CGAL_WEB_SFTP_HOST }} > ~/.ssh/known_hosts
      - name: rsync
        run: rsync -avz ./_site/ ${{ secrets.CGAL_WEB_SFTP_USERNAME }}@${{ vars.CGAL_WEB_SFTP_HOST }}:${{ vars.CGAL_WEB_SFTP_DESTDIR }}
```

---

I have actually pushed by suggestion directly.